api.planx.uk/modules/flows/findReplace/service.ts

```diff
@@ -84,8 +84,10 @@ const findAndReplaceInFlow = async (
   }
 
   // if matches, proceed with mutation to update flow data
-  const { client: $client } = getClient();
-  const response = await $client.request<UpdateFlow>(
+
+  // XXX: This is using the API Hasura role
```

Comment:
Let me know if a comment is not good enough here!

Alternatives considered are - 
 - Attempt a mutation before `UpdateFlow` which we know user has permission for (e.g. set `flow_slug` = `flow_slug`). If this works we know user should be allowed to update `flow.data`
 - Manually query permission (e.g. check join of userId → team_members → teamId → flowId). The tricky thing about a manual check is that this logic will always have to be updated alongside our permission and roles model.
 
 For now, we're checking if the user is a platformAdmin via middleware and then executing this mutation. This is totally fine I think - I just wanted to leave a cautionary note behind.

Reply:
Agree with your decision/approach here :heavy_plus_sign: thanks for explaining alternatives